server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -824,11 +824,11 @@ public IPAddressVO doInTransaction(TransactionStatus status) throws Insufficient
                     if (vlanDbIds == null || vlanDbIds.contains(nonDedicatedVlan.getId()))
                         nonDedicatedVlanDbIds.add(nonDedicatedVlan.getId());
                 }
-                if (dedicatedVlanDbIds != null && !dedicatedVlanDbIds.isEmpty()) {
+                if (!dedicatedVlanDbIds.isEmpty()) {
                     fetchFromDedicatedRange = true;
                     sc.setParameters("vlanId", dedicatedVlanDbIds.toArray());
                     errorMessage.append(", vlanId id=" + Arrays.toString(dedicatedVlanDbIds.toArray()));
-                } else if (nonDedicatedVlanDbIds != null && !nonDedicatedVlanDbIds.isEmpty()) {
```

Comment:
Null check not needed, the list has been created at line 793.

`List<Long> nonDedicatedVlanDbIds = new ArrayList<Long>();`

Reply:
yes, you could have told me before i checked it ;)